backend/api/pipeline.proto

```diff
@@ -105,26 +106,33 @@ service PipelineService {
     };
   }
 
+  // Get a pipeline version by pipeline version ID.
   rpc GetPipelineVersion(GetPipelineVersionRequest) returns (PipelineVersion) {
     option (google.api.http) = {
       get: "/apis/v1beta1/pipeline_versions/{version_id}"
     };
   }
 
+  // List all pipeline versions of a given pipeline.
   rpc ListPipelineVersions(ListPipelineVersionsRequest)
       returns (ListPipelineVersionsResponse) {
     option (google.api.http) = {
       get: "/apis/v1beta1/pipeline_versions"
     };
   }
 
+  // Delete a pipeline version by pipeline version ID. If the deleted pipeline
+  // version happens to be the default pipeline version of a certain pipeline,
+  // the default version of that pipeline will change to the latest pipeline
+  // version among all the remaining pipeline versions of that pipeline.
```

Comment:
discussion: what happens when the last version is deleted?

Reply:
Then the pipeline will have no version and default version is None. 
We allow pipeline having no pipeline versions under it. 
And that's also why we always encourage users to create run from pipeline versions instead of from pipeline. 
However, since users am familiar with the previous way of creating run by specify a pipeline, we try to be backward compatible, and use the default pipeline version if users specify pipeline when they should ideally specify a pipeline version.